packages/core/http/core-http-router-server-internal/src/versioned_router/oas_poc/generate_oas.ts

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+/**
+ * Heavily adapted version of https://github.com/jlalmes/trpc-openapi
+ */
+
+/* eslint-disable import/no-extraneous-dependencies */
+
+import type { OpenAPIV3 } from 'openapi-types';
+import zodToJsonSchema from 'zod-to-json-schema';
+import {
+  instanceofZodType,
+  instanceofZodTypeObject,
+  instanceofZodTypeLikeVoid,
+  instanceofZodTypeLikeString,
+  instanceofZodTypeOptional,
+  unwrapZodType,
+  zodSupportsCoerce,
+  instanceofZodTypeCoercible,
+  getPathParameters,
+} from './util';
+
+import type { CoreVersionedRouter } from '..';
+import type { VersionedRouterRoute } from '../types';
+
+export const openApiVersion = '3.0.0';
+
+export interface GenerateOpenApiDocumentOptions {
+  title: string;
+  description?: string;
+  version: string;
+  baseUrl: string;
+  docsUrl?: string;
```

Comment:
That sounds like a great idea! I'm not as familiar with that component, but if I understand correctly something like this will be needed once we start integrating this spec into existing docs?

Reply:
Yes, I think this type of support for linking would only be necessary if we want specifications to be able to link to each other (or other external websites).